__tests__/app.test.js

```diff
@@ -147,6 +147,151 @@ describe("app", () => {
+                expect.objectContaining({
+                  review_id: reviewId,
+                  title: expect.any(String),
+                  review_body: expect.any(String),
+                  designer: expect.any(String),
+                  review_img_url: expect.any(String),
+                  votes: 10,
+                  category: expect.any(String),
+                  owner: expect.any(String),
+                  created_at: expect.any(String),
+                })
+              );
+            });
+        });
+
+        test("200 - responds with updated review when decrease votes by passing negative inc_votes as a body object", () => {
+          const reviewId = 2;
+          const patchObject = {
+            inc_votes: -10,
+          };
+          return request(app)
+            .patch(`/api/reviews/${reviewId}`)
+            .send(patchObject)
+            .expect(200)
+            .then(({ body }) => {
+              const { review } = body;
+
+              expect(review).toEqual(
+                expect.objectContaining({
+                  review_id: reviewId,
+                  title: expect.any(String),
+                  review_body: expect.any(String),
+                  designer: expect.any(String),
+                  review_img_url: expect.any(String),
+                  votes: -5,
+                  category: expect.any(String),
+                  owner: expect.any(String),
+                  created_at: expect.any(String),
+                })
+              );
+            });
+        });
+      });
+
+      describe("Unsuccessful Responses", () => {
+        test("400 - response with message 'Bad request' when empty body passed in", () => {
+          const reviewId = 2;
+          const patchObject = "";
+          return request(app)
+            .patch(`/api/reviews/${reviewId}`)
+            .send(patchObject)
+            .expect(400)
+            .then(({ body }) => {
+              const { message } = body;
+
+              expect(message).toBe("Bad request");
+            });
+        });
+        test("400 - response with message 'Bad request' when empty body object passed in", () => {
+          const reviewId = 2;
+          const patchObject = {};
+          return request(app)
+            .patch(`/api/reviews/${reviewId}`)
+            .send(patchObject)
+            .expect(400)
+            .then(({ body }) => {
+              const { message } = body;
+
+              expect(message).toBe("Bad request");
+            });
+        });
+
+        test("400 - response with message 'Bad request' when review doesn't exist", () => {
+          const reviewId = 9999999;
+          const patchObject = "";
+          return request(app)
+            .patch(`/api/reviews/${reviewId}`)
+            .send(patchObject)
+            .expect(400)
+            .then(({ body }) => {
+              const { message } = body;
+
+              expect(message).toBe("Bad request");
+            });
+        });
+
```

Comment:
This one should be a 404 when a review is valid but doesn't exist.

Reply:
I've done changes. Please have a look. :-)